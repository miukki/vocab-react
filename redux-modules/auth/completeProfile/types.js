const prefix = 'auth/completeProfile'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_NAME')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('RESET_STATE')

export default types
