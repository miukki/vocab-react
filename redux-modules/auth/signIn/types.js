const prefix = 'auth/signIn'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_MOBILE')
types.set('CHANGE_PASSWORD')
types.set('SHOW_PASSWORD')
types.set('HIDE_PASSWORD')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('GO_FORGOT_PASSWORD')
types.set('RESET_STATE')

export default types
