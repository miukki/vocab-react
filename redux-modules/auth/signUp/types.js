const prefix = 'auth/signUp'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_MOBILE')
types.set('SEND_VCODE')
types.set('SEND_VCODE_DONE')
types.set('COUNTDOWN_VCODE')
types.set('CHANGE_VCODE')
types.set('CHANGE_PASSWORD')
types.set('SHOW_PASSWORD')
types.set('HIDE_PASSWORD')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('RESET_STATE')

export default types
