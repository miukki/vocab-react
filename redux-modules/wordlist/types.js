export const SELECT_ITEM = '@@wordlist/SELECT_ITEM'
