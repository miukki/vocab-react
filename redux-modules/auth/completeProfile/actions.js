import types from './types'

export const show = () => {
  return { type: types.SHOW }
}
export const hide = () => {
  return { type: types.HIDE }
}

export const changeName = name => {
  return { type: types.CHANGE_NAME, payload: { name } }
}

export const submit = () => {
  return { type: types.SUBMIT }
}

export const submitDone = () => {
  return { type: types.SUBMIT_DONE }
}

export const showError = errorType => {
  return { type: types.SHOW_ERROR, payload: { errorType } }
}

export const dismissError = () => {
  return { type: types.DISMISS_ERROR }
}
