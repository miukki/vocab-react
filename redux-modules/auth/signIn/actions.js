import types from './types'

export const show = () => {
  return { type: types.SHOW }
}
export const hide = () => {
  return { type: types.HIDE }
}

export const changeMobile = mobile => {
  return { type: types.CHANGE_MOBILE, payload: { mobile } }
}

export const changePassword = password => {
  return { type: types.CHANGE_PASSWORD, payload: { password } }
}

export const showPassword = () => {
  return { type: types.SHOW_PASSWORD }
}

export const hidePassword = () => {
  return { type: types.HIDE_PASSWORD }
}

export const submit = () => {
  return { type: types.SUBMIT }
}

export const submitDone = () => {
  return { type: types.SUBMIT_DONE }
}

export const showError = errorType => {
  return { type: types.SHOW_ERROR, payload: { errorType } }
}

export const dismissError = () => {
  return { type: types.DISMISS_ERROR }
}

export const goForgotPassword = () => {
  return { type: types.GO_FORGOT_PASSWORD }
}
