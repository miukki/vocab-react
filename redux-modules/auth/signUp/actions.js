import types from './types'

export const show = () => {
  return { type: types.SHOW }
}
export const hide = () => {
  return { type: types.HIDE }
}

export const changeMobile = mobile => {
  return { type: types.CHANGE_MOBILE, payload: { mobile } }
}

export const sendVcode = () => {
  return { type: types.SEND_VCODE }
}

export const sendVcodeDone = () => {
  return { type: types.SEND_VCODE_DONE }
}

export const countdownVcode = () => {
  return { type: types.COUNTDOWN_VCODE }
}

export const changeVcode = vcode => {
  return { type: types.CHANGE_VCODE, payload: { vcode } }
}

export const changePassword = password => {
  return { type: types.CHANGE_PASSWORD, payload: { password } }
}

export const showPassword = () => {
  return { type: types.SHOW_PASSWORD }
}

export const hidePassword = () => {
  return { type: types.HIDE_PASSWORD }
}

export const submit = () => {
  return { type: types.SUBMIT }
}

export const submitDone = () => {
  return { type: types.SUBMIT_DONE }
}

export const showError = errorType => {
  return { type: types.SHOW_ERROR, payload: { errorType } }
}

export const dismissError = () => {
  return { type: types.DISMISS_ERROR }
}
