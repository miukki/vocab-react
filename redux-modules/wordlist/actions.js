import * as types from './types'

export const selectItem = index => {
  return {
    type: types.SELECT_ITEM,
    index
  }
}
