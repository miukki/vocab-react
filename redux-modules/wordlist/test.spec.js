import Expo from 'expo'
import { fromJS, Map, List } from 'immutable'
import * as matchers from 'jest-immutable-matchers'
import reducer from './index'
import * as types from './types'
import { selectItem } from './actions'
import wordsLists from '../../data/words_lists.js'

const initialState = fromJS({
  items: [...wordsLists.keys()].map((list, index) => {
    return {
      title: list,
      onPress: () => {
        window.store.dispatch(selectItem(index))
      },
      progress: 10 * index
    }
  })
})

describe('wordListPicker reducer', () => {
  beforeEach(() => {
    jest.addMatchers(matchers)
  })

  it('should return the initial state', () => {
    expect(JSON.stringify(reducer(undefined, {}))).toMatchSnapshot()
    // use JSON.stringify to avoid test fail with “Compared values have no visual difference.”
    // TODO: find a elegance way to compare
  })

  it('should handle SELECT_ITEM', () => {
    expect(
      reducer(initialState, {
        type: types.SELECT_ITEM,
        index: 1
      })
    ).toMatchSnapshot()
  })
})
