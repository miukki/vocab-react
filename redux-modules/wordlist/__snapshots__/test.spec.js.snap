// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wordListPicker reducer should handle SELECT_ITEM 1`] = `
Immutable.Map {
  items: Immutable.List [
    Immutable.Map {
      title: "IELTS Listening",
      onPress: [Function],
      progress: 0,
    },
    Immutable.Map {
      title: "IELTS Reading",
      onPress: [Function],
      progress: 10,
    },
    Immutable.Map {
      title: "IELTS Writing",
      onPress: [Function],
      progress: 20,
    },
    Immutable.Map {
      title: "IELTS Speaking",
      onPress: [Function],
      progress: 30,
    },
    Immutable.Map {
      title: "TOEFL Listening",
      onPress: [Function],
      progress: 40,
    },
    Immutable.Map {
      title: "TOEFL Reading",
      onPress: [Function],
      progress: 50,
    },
    Immutable.Map {
      title: "TOEFL Writing",
      onPress: [Function],
      progress: 60,
    },
    Immutable.Map {
      title: "TOEFL Speaking",
      onPress: [Function],
      progress: 70,
    },
  ],
}
`;

exports[`wordListPicker reducer should return the initial state 1`] = `"{\\"pressable\\":false,\\"items\\":[]}"`;
